core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -191,7 +191,20 @@ private[spark] class PythonWorkerFactory(pythonExec: String, envVars: Map[String
         daemon = pb.start()
 
         val in = new DataInputStream(daemon.getInputStream)
-        daemonPort = in.readInt()
+        try {
+          daemonPort = in.readInt()
+        } catch {
+          case exc: EOFException =>
+            throw new IOException(s"No port number in $daemonModule's stdout")
+        }
+
+        // test that the returned port number is within a valid range.
+        // note: this does not cover the case where the port number
+        // is arbitrary data but is also coincidentally within range
+        if (daemonPort < 1 || daemonPort > 0xffff) {
+          throw new IOException(s"Bad port number in $daemonModule's stdout: " +
+            f"0x$daemonPort%08x")
```

Comment:
Shall we also show the actual number?

Reply:
just a thought:

 this error message won't mean much to the typical user.  Would it be sensible to tell the user exactly what python command to run themselves to figure out the problem?  Eg. "unexpected stdout from /foo/bar/some/path/to/python -m /path/to/daemon.py".  That's what would help with that sitecustomization.py case.  Or not useful in general?

Reply:
Yes, that makes sense. I might not be able to get it as clear as the exact path to the file, since the PythonWorkerFactory sets a PYTHONPATH environmental variable and then lets python itself figure out where on those paths the module actually lives. But I could tell the user how the PYTHONPATH was set up (in a generic sense, without using any shell's syntax) and then how the python command was subsequently run.